pkg/controller/pravegacluster/upgrade.go

```diff
@@ -587,6 +601,9 @@ func (r *ReconcilePravegaCluster) scaleSegmentStoreSTS(p *pravegav1beta1.Pravega
 		return fmt.Errorf("updating statefulset (%s) failed due to %v", newsts.Name, err)
 	}
 	*oldsts.Spec.Replicas = *oldsts.Spec.Replicas - 2
+	currentSts := &appsv1.StatefulSet{}
+
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: oldsts.Name, Namespace: p.Namespace}, currentSts)
```

Comment:
Is there a reason for the last two lines of code?
The error is not processed (`err` variable is reassigned on the next line), `currentSts` does not seem to be used anywhere.

Reply:
Will remove it , added for debugging

Reply:
Thanks for finding it, removed it now